plugins/worker/lib/BlockedBy.js

```diff
@@ -41,12 +41,16 @@ async function collapseBuilds({ waitingKey, buildId, blockingBuildIds }) {
 
             // if the build is no longer in the waiting queue, don't collapse it
             if (count > 0) {
-                await helper.updateBuildStatus({
-                    redisInstance: this.queueObject.connection.redis,
-                    buildId: bId,
-                    status: 'COLLAPSED',
-                    statusMessage: `Collapsed to build: ${buildId}`
-                });
+                await helper
+                    .updateBuildStatus({
+                        redisInstance: this.queueObject.connection.redis,
+                        buildId: bId,
+                        status: 'COLLAPSED',
+                        statusMessage: `Collapsed to build: ${buildId}`
+                    })
+                    .catch(err => {
```

Comment:
As I mentioned in the [comments](https://github.com/screwdriver-cd/screwdriver/issues/2503#issuecomment-874594263) for details, the `err` returned by [`updateBuildStatus`](https://github.com/screwdriver-cd/queue-service/blob/deb923bed0274f3bd4c32253b5bb26a8e1aa2db7/plugins/helper.js#L70) may be `null`.
If `res.statusCode` is other than 200 and `err` is null, it is better to return some meaningful err.

Reply:
I added an error handle for non 200 statusCode to updateBuildStatus.